File: packages/types/src/game.ts
export interface Game {
  id: number;
  steam_app_id: number | null;
  igdb_id: number | null;
  name: string;
  release_date: string | null;
  developer: string | null;
  publisher: string | null;
  cover_url: string | null;
  platforms: string | null;
  genre: string | null;
  steam_deck_status: string | null;
  controller_support: string | null;
}

export interface Review {
  id: number;
  review_id: string;
  game_id: number;
  review_text: string | null;
  recommended: boolean;
  playtime: number | null;
  created_at: string | null;
}

export interface PlayerMetric {
  id: number;
  game_id: number;
  current_players: number;
  peak_players: number;
  retention_30d: number | null;
  retention_90d: number | null;
  recorded_at: string;
}

export interface HealthScore {
  id: number;
  game_id: number;
  score: number;
  sentiment: number;
  stability: number;
  retention: number;
  developer_support: number;
  recommendation: string;
  created_at: string;
}

export interface IssueReport {
  id: number;
  game_id: number;
  review_id: number | null;
  issue_type: string;
  confidence: number;
}

export interface Patch {
  id: number;
  game_id: number;
  version: string | null;
  title: string | null;
  notes: string | null;
  bug_fixes: string | null;
  performance_fixes: string | null;
  server_fixes: string | null;
  released_at: string | null;
}

File: apps/api/src/models/game.py
from datetime import date
from sqlalchemy import Integer, String, Date
from sqlalchemy.orm import Mapped, mapped_column, relationship

from src.database import Base


class Game(Base):
    __tablename__ = "games"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    steam_app_id: Mapped[int | None] = mapped_column(Integer, unique=True, nullable=True)
    igdb_id: Mapped[int | None] = mapped_column(Integer, unique=True, nullable=True)
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    release_date: Mapped[date | None] = mapped_column(Date, nullable=True)
    developer: Mapped[str | None] = mapped_column(String(255), nullable=True)
    publisher: Mapped[str | None] = mapped_column(String(255), nullable=True)
    cover_url: Mapped[str | None] = mapped_column(String(512), nullable=True)
    platforms: Mapped[str | None] = mapped_column(String(512), nullable=True)
    genre: Mapped[str | None] = mapped_column(String(255), nullable=True)
    controller_support: Mapped[str | None] = mapped_column(String(50), nullable=True)
    steam_deck_status: Mapped[str | None] = mapped_column(String(50), nullable=True)

    reviews = relationship("Review", back_populates="game")
    player_metrics = relationship("PlayerMetric", back_populates="game")
    health_scores = relationship("HealthScore", back_populates="game")
    issue_reports = relationship("IssueReport", back_populates="game")
    patches = relationship("Patch", back_populates="game")
